intellij/src/main/java/services/kiotaVersionHandler.java

```diff
@@ -0,0 +1,144 @@
+package services;
+
+import com.thetransactioncompany.jsonrpc2.*;
+
+// The JSON-RPC 2.0 server framework package
+import com.thetransactioncompany.jsonrpc2.server.*;
+
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class kiotaVersionHandler implements RequestHandler {
+
+    @Override
+    public String[] handledRequests() {
+        return new String[]{"getkiotaversion"};
+    }
+
+    @Override
+    public JSONRPC2Response process(JSONRPC2Request request, MessageContext requestCtx) {
+        if (request.getMethod().equals("getkiotaversion")) {
+            // Echo first parameter
+            //ArrayList<String> response = new ArrayList<String>();
+
+            String version = getkiotaversion();
+            return new JSONRPC2Response(version, request.getID());
+        } else {
+            // Method name not supported
+            return new JSONRPC2Response(JSONRPC2Error.METHOD_NOT_FOUND, request.getID());
+        }
+    }
+
+    public String getkiotaversion() {
+        String software = "kiota";
+
+        try {
+            Process process = new ProcessBuilder(software, "--version").start();
```

Comment:
you should start the process with the rpc command instead to get the json RPC server
https://github.com/microsoft/kiota/blob/293a9625113dd7015ddef286ce579d8b7cf2d92e/vscode/microsoft-kiota/src/kiotaInterop.ts#L9